### DISCUSSION TITLE
Can't get causer name

### DISCUSSION THREAD

**Discussion by Claudiu5** [spatie/laravel-activitylog#789] [Q&A] [answered]
I'm using this spatie activitylog package, that is very helpful . But I create a user activity feed that only admin can see , but for that i have used permissions, anyway, my questions is how do i get users name instead of users id. In my view table is like
```
<th>User ID </th>
<th>User Name </th>

@foreach ( activity_log as $feed)
   <td>{{$feed->causer_id}}</td> <br>

   <td> {{ what should i write or what metod should be for get the name of user who did the action }} </td>
@endforeach
```
I have to say that the ID is not really necessary, what am I interest to is the name.

**Comment by Gummibeer** [comment, marked as answer]
You can use the `causer` relationship. For example `{{ $activity->causer?->name }}`

**Reply by mrkazmierczakuw** [reply]
Could you explain this?

**Reply by mrkazmierczakuw** [reply]
There is no reference on migration, because it's $table->nullableMorphs('causer', 'causer');

**Reply by DDaniaLL** [reply]
you have to use the method causer like this : 
**{{ optional($activity->causer)->name }}**
it works Perfectly.

**Reply by ouaziz** [reply]
it works for laravel 11
